datalad/core/local/save.py

```diff
@@ -196,7 +196,14 @@ def __call__(path=None, message=None, dataset=None,
                 untracked=untracked_mode,
                 recursive=recursive,
                 recursion_limit=recursion_limit,
+                on_failure='ignore',
```

Comment:
With the conditional below, this change should not be needed, as the error handling will be done outside. Moreover, without `status(on_failure=not ignore)` setting, but with `save(on_failure='continue')` wouldn't it bring `save()` to a halt (by means of `status()` aborting), although that it not desired by the user?

Reply:
> With the conditional below, this change should not be needed, as the error handling will be done outside.

The purpose of  `status(..., on_failure="ignore")`is to let the handling be done outside.  If we don't do it, the failure will happen at the status call itself, meaning the on_failure value given to save() isn't honored.  To use the CLI-version of the test from this PR, here's how `--on-failure=continue save` behaves before and after this patch:

```sh
#!/bin/sh

cd $(mktemp -dt dl-XXXXXXX)
touch outside
datalad create ds && cd ds && touch within
datalad --on-failure=continue save  ../outside within
```

Before:

```
[INFO   ] Creating a new annex repo at /tmp/dl-4v81AUB/ds 
create(ok): /tmp/dl-4v81AUB/ds (dataset)
[ERROR  ] path not underneath this dataset [status(/tmp/dl-4v81AUB/outside)]
```

After:

```
[INFO   ] Creating a new annex repo at /tmp/dl-qX9sNhx/ds 
create(ok): /tmp/dl-qX9sNhx/ds (dataset)
[ERROR  ] path not underneath this dataset [status(/tmp/dl-qX9sNhx/outside)] 
status(error): ../outside [path not underneath this dataset]
add(ok): within (file)
save(ok): . (dataset)
action summary:
  add (ok: 1)
  save (ok: 1)
  status (error: 1)
```

Reply:
Oh, sorry for wasting your time. I am not sure why, but I saw it removing the line, not adding it. Sorry...